src/Features/JsonPatch/test/JsonPatchDocumentJObjectTest.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Dynamic;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.JsonPatch
+{
+    public class JsonPatchDocumentJObjectTest
+    {
+        [Fact]
+        public void ApplyTo_ExpandoObject_AddAndReplace()
+        {
+            // Arrange
+            var list = new List<string>
+            {
+                "[{ 'op': 'add', 'path': '/customData', 'value': {'bar': 'baz'}}, { 'op': 'replace', 'path': '/customData/bar', 'value': 'bazz' }]"
+            };
+
+            dynamic model = new ExpandoObject();
```

Comment:
Maybe I'm missing something - why is `ExpandoObject` relevant here?

Reply:
My mistake - I misinterpreted the issue from the OP and thought there was an issue with ExpandoObject in combination with JObject. But after re-reading, it seems that he was simply using ExpandoObject to work around the issue due to the lack of an JObjectAdapter.

I’ll get this test removed 👍🏻

Reply:
OK great.

Reply:
Send a ping when this is done and I'll see if I can merge it. We've been having some CI issues today which is why the builds are red.

Reply:
@rynowak All set.